package/MDAnalysis/topology/tpr/utils.py

```diff
@@ -284,7 +286,7 @@ def extract_box_info(data, fver):
     return obj.Box(box, box_rel, box_v)
 
 
-def do_mtop(data, fver):
+def do_mtop(data, fver, tpr_resid_from_one=True):
```

Comment:
I feel the default should remain `False` because that is what the TPR actually contains. IMHO the TPR functions themselves could/should stay as faithful to the original format as possible. Assume someone wrote their own TPR parser with our functions. With the new default you break their code needlessly, with the old defaults nothing changes and our own parser just makes the well-documented choice to index from 1.

Reply:
I can be convinced otherwise (ie to let default `True` stand) but wanted to discuss this point first before approving.

Reply:
Others may want to chime in (maybe the most knowledgeable, @jbarnoud) but I agree with your argument.

Reply:
I am going to approve because it looks sensible from my point but let @jbarnoud also say something before merging.

Reply:
TPR indeed index from 0, but it is a bit moot because you do not read a TPR directly. Instead, you end up reading your gro and ITP files to make sense of your TPR so consistency there seems the right thing to do.

Reply:
My previous objection was that it would break the existing code, but the deprecation in 1.1.0 should be enough.